lamindb/_collection.py

```diff
@@ -221,6 +222,33 @@ def from_artifacts(artifacts: Iterable[Artifact]) -> tuple[str, dict[str, str]]:
     return hash
 
 
+# docstring handled through attach_func_to_class_method
+def open(self, is_run_input: bool | None = None) -> PyArrowDataset:
+    if self._state.adding:
+        artifacts = self._artifacts
+        logger.warning("The collection isn't saved, consider calling `.save()`")
+    else:
+        artifacts = self.ordered_artifacts.all()
+    paths = [artifact.path for artifact in artifacts]
+    # this checks that the filesystem is the same for all paths
+    # this is a requirement of pyarrow.dataset.dataset
+    fs = paths[0].fs
+    for path in paths[1:]:
+        # this assumes that the filesystems are cached by fsspec
+        if path.fs is not fs:
+            raise ValueError(
+                "The collection has artifacts with different filesystems, this is not supported."
+            )
+    if not _is_pyarrow_dataset(paths):
+        raise ValueError(
+            "This collection is not compatible with pyarrow.dataset.dataset."
+        )
+    access = _open_pyarrow_dataset(paths)
```

Comment:
I'd call this `dataset`, but no big deal.

Reply:
done